src/Core/Services/ConfigService.cs

```diff
@@ -49,14 +49,43 @@ public async Task<ConfigResponse> GetAsync(bool forceRefresh = false)
             return _configs;
         }
 
-        public async Task<bool> GetFeatureFlagAsync(string key, bool forceRefresh = false)
+        public async Task<bool> GetFeatureFlagBoolAsync(string key, bool forceRefresh = false, bool defaultValue = false)
         {
             await GetAsync(forceRefresh);
-            return _configs?.FeatureStates?.TryGetValue(key, out var val) == true
+            if (_configs?.FeatureStates?.TryGetValue(key, out var val) == true
                     &&
-                    val is bool boolVal
+                    val is bool boolVal)
+            {
+                return boolVal;
+            }
+
+            return defaultValue;
+        }
+
+        public async Task<string> GetFeatureFlagStringAsync(string key, bool forceRefresh = false, string defaultValue = null)
+        {
+            await GetAsync(forceRefresh);
+            if (_configs?.FeatureStates?.TryGetValue(key, out var val) == true
+                    &&
+                    val is string strVal)
+            {
+                return strVal;
+            }
+
+            return defaultValue;
+        }
+
+        public async Task<int> GetFeatureFlagIntAsync(string key, bool forceRefresh = false, int defaultValue = 0)
+        {
+            await GetAsync(forceRefresh);
+            if (_configs?.FeatureStates?.TryGetValue(key, out var val) == true
                     &&
-                    boolVal;
+                    val is int intVal)
+            {
+                return intVal;
+            }
```

Comment:
🎨  Instead of having different methods for each type, you can have just one and use generics:
```c#
public async Task<T> GetFeatureFlagAsync<T>(string key, bool forceRefresh = false, T defaultValue = default)
{
    await GetAsync(forceRefresh);
    if (_configs?.FeatureStates?.TryGetValue(key, out var val) == true
            &&
            val is T actualValue)
    {
        return actualValue;
    }

    return defaultValue;
}
```

and call it like:
```c#
_displayEuEnvironment = await _configService.GetFeatureFlagAsync<bool>(ConfigResponse.DisplayEuEnvironmentFlag, forceRefresh: true);
```

Reply:
I was doing it like that but i changed it on purpose to lock it to the 3 types available server side, and for the implementation to be similar.

Reply:
Alright I think we can make the generic method private and leave the three public calls restricted, so we expose the restricted version but we have the generic one to avoid repeating code

Reply:
sounds good! 👍